src/base/dimensionmanager.js

```diff
@@ -29,7 +29,13 @@ const DimensionManagerPrototype = {
     // - no overlapping dimensionModels between markers or changing dimension models is fine
     // - one entities model per dimension
     // - only one time model (state.time)
-    const oldSpaceEntities = Object.keys(oldSpace).filter(dim => oldSpace[dim]._type == "entities");
+    const oldSpaceEntities = Object.keys(oldSpace).filter(modelName => oldSpace[modelName]._type == "entities" && !newSpaceDimensions.includes(oldSpace[modelName].dim));
+    const savedEntities = Object.keys(oldSpace).reduce((result, modelName) => {
```

Comment:
these two procedures were very hard to understand, maybe better to rewrite in such a way that:
- a similar way is used for both actions since they do similar things
- use a dedicated function with a descriptive name
- use descriptive set terminology, like: "intersection" and "substraction"

Reply:
For example:
unusedOldEntities = oldSpace - newSpace // difference/substraction
reusedOldEntities = oldSpace & newSpace // intersection